src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -82,7 +82,7 @@ public SchemaObjectType(ObjectSchema objectSchema, BuildContext context)
         private bool _hasCalculatedDefaultDerivedType;
         public SerializableObjectType? DefaultDerivedType => _defaultDerivedType ??= BuildDefaultDerivedType();
 
-        protected override bool IsAbstract => !Configuration.SuppressAbstractBaseClasses.Contains(DefaultName) && ObjectSchema.Discriminator?.All != null && ObjectSchema.Parents?.All.Count == 0;
+        protected override bool IsAbstract => !Configuration.SuppressAbstractBaseClasses.Contains(DefaultName) && ObjectSchema.Discriminator?.All != null && ObjectSchema.DiscriminatorValue == null;
```

Comment:
we make a class abstract when it is the base class in the discriminated set.
Here we check if it is a base type by checking if it has no parent models, this is true for very limited scenarios.
This PR changes it to check the discriminatorValue is null because the base model in the set does not have the value, while its derived types have.

Reply:
Is empty string a possible value for an abstract base class return from service side?

Reply:
no. The service does not accept a "type", it accepts a json with a valid discriminator value. If the service team does not listed empty string as a valid discriminator value, it does work. Therefore we should not do that.